### PR TITLE
minor fix for aggregate_task_reports

### PR DESCRIPTION
**Pull request by dexterju**
**Patch description**
The current aggregate_task_reports breaks the old imagechat agent where it will generate a metrics of style imagechat{{first_round:hit@1, second_round:}} etc

**Testing steps**
Test with comment_battle agent
**Logs**
```
{'exs': 15000, 'accuracy': 0.0102, 'f1': 0.0736, 'hits@1': 0.0102, 'hits@5': 0.0519, 'hits@10': 0.101, 'hits@100': 1.0, 'bleu': 0.009928, 'rouge-1': 0.08521, 'rouge-2': 0.01487, 'rouge-L': 0.07907, 'first_rou
nd': {'hitsAt1KC100': 0.01, 'loss': -0.5, 'med_rank': 50.0}, 'second_round': {'hitsAt1KC100': 0.0118, 'loss': -0.5, 'med_rank': 49.0}, 'third_round': {'hitsAt1KC100': 0.0088, 'loss': -0.5, 'med_rank': 50.5}}
{'exs': 5000, 'accuracy': 0.0412, 'f1': 0.1728, 'hits@1': 0.0412, 'hits@5': 0.178, 'hits@10': 0.321, 'hits@100': 1.0, 'bleu': 0.04148, 'rouge-1': 0.1809, 'rouge-2': 0.05158, 'rouge-L': 0.1728, 'first_round':
{'hitsAt1KC100': 0.0412, 'loss': -0.5, 'med_rank': 20.0}, 'second_round': {}, 'third_round': {}}
{'exs': 1014, 'accuracy': 0.03649, 'f1': 0.191, 'hits@1': 0.0365, 'hits@5': 0.179, 'hits@10': 0.323, 'hits@100': 1.0, 'bleu': 0.03649, 'rouge-1': 0.2062, 'rouge-2': 0.05138, 'rouge-L': 0.1928, 'first_round':
{'hitsAt1KC100': 0.03649, 'loss': -0.5, 'med_rank': 18.0}, 'second_round': {}, 'third_round': {}}
valid:{'tasks': {'comment_battle:imageDialog': {'exs': 15000, 'accuracy': 0.0102, 'f1': 0.0736, 'hits@1': 0.0102, 'hits@5': 0.0519, 'hits@10': 0.101, 'hits@100': 1.0, 'bleu': 0.009928, 'rouge-1': 0.08521, 'ro
uge-2': 0.01487, 'rouge-L': 0.07907, 'first_round': {'hitsAt1KC100': 0.01, 'loss': -0.5, 'med_rank': 50.0}, 'second_round': {'hitsAt1KC100': 0.0118, 'loss': -0.5, 'med_rank': 49.0}, 'third_round': {'hitsAt1KC
100': 0.0088, 'loss': -0.5, 'med_rank': 50.5}}, 'internal:coco_caption:PersonalityTeacher': {'exs': 5000, 'accuracy': 0.0412, 'f1': 0.1728, 'hits@1': 0.0412, 'hits@5': 0.178, 'hits@10': 0.321, 'hits@100': 1.0
, 'bleu': 0.04148, 'rouge-1': 0.1809, 'rouge-2': 0.05158, 'rouge-L': 0.1728, 'first_round': {'hitsAt1KC100': 0.0412, 'loss': -0.5, 'med_rank': 20.0}, 'second_round': {}, 'third_round': {}}, 'internal:flickr30
k:PersonalityTeacher': {'exs': 1014, 'accuracy': 0.03649, 'f1': 0.191, 'hits@1': 0.0365, 'hits@5': 0.179, 'hits@10': 0.323, 'hits@100': 1.0, 'bleu': 0.03649, 'rouge-1': 0.2062, 'rouge-2': 0.05138, 'rouge-L':
0.1928, 'first_round': {'hitsAt1KC100': 0.03649, 'loss': -0.5, 'med_rank': 18.0}, 'second_round': {}, 'third_round': {}}}, 'exs': 21014, 'accuracy': 0.01884, 'f1': 0.1029, 'hits@1': 0.01885, 'hits@5': 0.08804
, 'hits@10': 0.1641, 'hits@100': 1.0, 'bleu': 0.01872, 'rouge-1': 0.1138, 'rouge-2': 0.02537, 'rouge-L': 0.1069, 'first_round': {'comment_battle:imageDialog': {'hitsAt1KC100': 0.01, 'loss': -0.5, 'med_rank':
50.0}, 'internal:coco_caption:PersonalityTeacher': {'hitsAt1KC100': 0.0412, 'loss': -0.5, 'med_rank': 20.0}, 'internal:flickr30k:PersonalityTeacher': {'hitsAt1KC100': 0.03649, 'loss': -0.5, 'med_rank': 18.0}}
, 'second_round': {'comment_battle:imageDialog': {'hitsAt1KC100': 0.0118, 'loss': -0.5, 'med_rank': 49.0}, 'internal:coco_caption:PersonalityTeacher': {}, 'internal:flickr30k:PersonalityTeacher': {}}, 'third_
round': {'comment_battle:imageDialog': {'hitsAt1KC100': 0.0088, 'loss': -0.5, 'med_rank': 50.5}, 'internal:coco_caption:PersonalityTeacher': {}, 'internal:flickr30k:PersonalityTeacher': {}}, 'warning': 'metri
cs are averaged across tasks and weighted by the number of examples per task'}
```

### CODE REVIEW

File: parlai/core/metrics.py [hunk 1]
```diff
@@ -82,14 +83,17 @@ def aggregate_task_reports(reports, tasks, micro=True):
...
+        if all([isinstance(v, Number) for v in task_vals.values()]):
+            if micro:
+                # average over the number of examples
+                vals = [task_vals[task] * exs[task] for task in tasks]
+                total_report[metric] = round_sigfigs(sum(vals) / total_exs, 4)
+            else:  # macro
+                # average over tasks
+                vals = task_vals.values()
+                total_report[metric] = round_sigfigs(sum(vals) / len(vals), 4)
+        else:
```
Comment (emilydinan) [inline parlai/core/metrics.py]: i don't think this is the right behavior here. if it's a number, let's aggregate it, otherwise let's ignore it (no "else")

Comment (klshuster) [reply parlai/core/metrics.py]: For the image chat model, some of the `task_vals` are going to be dicts corresponding to which round is being computed; e.g., the metrics dict might look like the following:

```
metrics = {
   'first_round': {
          'hits@1': ...
           ...
     },
     'second_round': {...},
     'third_round': {...},
}
```

Comment (dexterju) [reply parlai/core/metrics.py]: As far as these separated metrics for each task still get printed we can get rid of the else. Just want to make sure we don't lose information here : )

Comment (emilydinan) [reply parlai/core/metrics.py]: as @dexterju said, they will still be printed, they just won't be aggregated across different tasks. is this ok @klshuster ?

Comment (klshuster) [reply parlai/core/metrics.py]: ahh yeah should be fine

**Comment by dexterju** [comment]
Else removed!

**Pull request merged**